src/main/java/org/opentripplanner/api/parameter/QualifiedModeSet.java

```diff
@@ -122,7 +122,7 @@ public RequestModes getRequestModes() {
                         accessMode = StreetMode.CAR_TO_PARK;
                         transferMode = StreetMode.WALK;
                         egressMode = StreetMode.WALK;
-                        directMode = StreetMode.CAR_TO_PARK;
+                        directMode = null;
```

Comment:
This makes it so that it's not possible to ever look for car park itineraries with no transit?

Reply:
This disables it if you use the Legacy GraphQL and REST APIs and set a mode of `CAR_PARK`. You can use the Transmodel API to set any direct_mode you want.

Reply:
I don't at least mind that it would no longer by possible with the legacy GraphQL API. Not sure if that should still remain a possibility for the REST API. Could we make it so that if no transit modes are specified in the query, it would enable the directMode search?

Reply:
I would prefer the existing functionality being preserved, with the `directMode` being set. Currently `mode=BICYCLE_RENTAL` returns a bicycle rental only trip. I would expect `mode=CAR_PARK` to behave similarly.

Reply:
I get where you're coming from but in my tests computing the direct mode CAR_PARK was often the slowest part of the entire query. It appears to try to leave the car on all stations along the way and then calculate a walking leg from there.

Reply:
@flaktack How would you feel about CAR_PARK being used as a direct mode only if there is no transit mode set? Otherwise I suggest CAR.

The reason for this is that CAR_PARK is often up to three times slower than the rest of the transit search.

Reply:
I would prefer always keeping the direct mode -- regardless of transit modes being set. For now it preserves the existing functionality, is in line with how the other modes work in the rest api, and is easier to understand/document.

This currently only affects the rest api along with the legacy graphql api: an alternative option would be to instead extend either of these apis with an option to included/exclude the direct mode when planning a trip? That would both be backward-compatible, easy to explain/document and can be set as-needed.

Presumably in the new API this wouldn't be needed anymore, since the values would be explicitly set instead of using heuristics.

Reply:
Even if it's terrible for performance? In some of my tests it takes 3 times as long.

Reply:
@flaktack Hannes has managed to give CAR_PARK a huge speed boost by modifying the A* heuristic: https://github.com/opentripplanner/OpenTripPlanner/pull/3906/commits/ee08ebe86a2f01dceb8f8992c5d77724a3a5c523

Reply:
Seemingly it also affects vehicle rental but doesn't affect the search when arriveBy=true?

Reply:
We could add a check that if it is arrveBy, then use walk speed when the rental state is `BEFORE_RENTING`